apps/passport-client/components/screens/ProveScreen/SemaphoreSignatureProveScreen.tsx

```diff
@@ -103,11 +109,19 @@ export function SemaphoreSignatureProveScreen({
 async function fillArgs(
   identity: Identity,
   uuid: string,
-  modifiedArgs: SemaphoreSignaturePCDArgs
+  modifiedArgs: SemaphoreSignaturePCDArgs,
+  options?: ProveOptions
 ): Promise<SemaphoreSignaturePCDArgs> {
   const signedMessage = modifiedArgs.signedMessage;
 
-  if (signedMessage.value === undefined) {
+  if (options?.signIn) {
+    console.log("this signature request is for signing into a website", uuid);
+    const payload: SignInMessagePayload = {
+      uuid,
+      referrer: getReferrerHost(),
```

Comment:
```
origin: getReferrerOrigin()
```

a detail but host is `zupoll.org` for display, origin is `https://zupoll.org` is what all the security stuff (Same-Origin Policy, cookies/localStorage, etc) keys off of

Reply:
Good catch, sorry about that.